src/Catel.Core/Catel.Core.Shared/WeakReferences/WeakEventHandler.cs

```diff
@@ -0,0 +1,173 @@
+﻿namespace Catel
```

Comment:
I thought that the WeakEventHandler also has non-typed methods? If not, we should extend that one instead of introducing a new class.